src/plugins/teleop/Teleop.qml

```diff
@@ -0,0 +1,399 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+import QtQuick 2.9
+import QtQuick.Controls 2.2
+import QtQuick.Controls.Material 2.1
+import QtQuick.Controls.Styles 1.4
+import QtQuick.Layouts 1.3
+
+Rectangle {
+  color:"transparent"
+  Layout.minimumWidth: 300
+  Layout.minimumHeight: 900
+  anchors.fill: parent
+  focus: true
+
+  // Topic input
+  Label {
+    id: topicLabel
+    text: "Topic:"
+    anchors.top: parent.top
+    anchors.topMargin: 10
+    anchors.left: parent.left
+    anchors.leftMargin: 5
+  }
+  TextField {
+    id: topicField
+    anchors.top: topicLabel.bottom
+    anchors.topMargin: 5
+    anchors.left: parent.left
+    anchors.leftMargin: 5
+    Layout.fillWidth: true
+    text:"/cmd_vel"
+    placeholderText: qsTr("Topic to publish...")
+    onEditingFinished: {
+      Teleop.OnTopicSelection(text)
+    }
+  }
+
+  // Linear velocity input
+  Label {
+    id: linearVelLabel
+    text: "Linear:"
+    anchors.top: topicField.bottom
+    anchors.topMargin: 10
+    anchors.left: parent.left
+    anchors.leftMargin: 5
+  }
+  TextField {
```

Comment:
How about using `IgnSpin` here, which has up and down arrows to choose the value? It also has the advantage of only allowing numbers.

Reply:
Addressed